### PR TITLE
MDX to TS init

### PR DESCRIPTION
**Pull request by damianmef**

### CODE REVIEW

File: lib/js/components/Badges/Badge/Badge.stories.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+} as Meta<typeof Badge>;
+
+const Template: Story<{ label: string; iconUrl: string; color: Object }> = (argTypes) => ({
+	components: { Badge },
+	props: Object.keys(argTypes),
+	template:
+		'<div style="display: flex"><Badge :color="color" :label="label"><img alt="Badge" style="width: 100%; height: 100%" :src="iconUrl" /></Badge></div>',
+});
+
+export const Primary = Template.bind({});
```
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:20]: why do we call it `Primary`?

File: lib/js/components/Badges/Badge/Badge.stories.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { BADGE_COLORS } from './Badge.consts';
+
+import { Meta, Story } from '@storybook/vue';
+
+export default {
+	title: 'Components/Badges/Badge',
+	component: Badge,
+} as Meta<typeof Badge>;
+
+const Template: Story<{ label: string; iconUrl: string; color: Object }> = (argTypes) => ({
```
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:13]: is it ok that we call it `Template` while it's `Story` type?

File: lib/js/components/Icon/Icon.consts.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ const BETHINK_ICONS = {
 	RIBBON: Ribbon,
 } as const;
 
-export const ICONS = {
+export let ICONS = {
```
Comment (kvas-damian) [inline lib/js/components/Icon/Icon.consts.ts:21]: why this change?

File: lib/js/components/Badges/Badge/Badge.stories.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// Badge.stories.ts
+
```
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:2]:
```suggestion
```

File: lib/js/components/Badges/Badge/Badge.stories.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		defaultValue: BADGE_COLORS.NEUTRAL,
+	},
+};
+
+Primary.parameters = {
+	design: {
+		type: 'figma',
+		url: 'https://www.figma.com/file/izQdYyiBR1GQgFkaOIfIJI/LMS---DS---Components?node-id=1552%3A34953',
+	},
+};
```
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:39]: Can we remove `Badge.stories.mdx` now?

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
+// Button.stories.ts
+
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:2]:
```suggestion
```

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+import { Meta, Story } from '@storybook/vue';
+
+export default {
+	title: 'Components/Buttons/Button',
+	component: Button,
+} as Meta<typeof Button>;
+
+const Template: Story<{
+	slotText: Object;
+	size: Object;
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:23]: incorrect type for many props in this list

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+	components: { Button },
+	props: Object.keys(argTypes),
+	template: `<div :class="{ contrastBackground: isInverted }">
+                <Button
+                  :size="size"
+                  :type="type"
+                  :state="state"
+                  :radius="radius"
+                  :color="color"
+                  :elevation="elevation"
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:41]: did you consider using `v-bind=$props` syntax? I'm not sure what are dowsnsides of it, but it looks promising

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+import { ICONS } from '../../Icon';
+
+import { Meta, Story } from '@storybook/vue';
+
+export default {
+	title: 'Components/Buttons/Button',
+	component: Button,
+} as Meta<typeof Button>;
+
+const Template: Story<{
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:21]: I see in docs that Storybook recommends `StoryFn()`: https://storybook.js.org/docs/vue/writing-stories/introduction
What was the reason to not follow that rule?

File: lib/js/components/Badges/Badge/Badge.stories.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+Primary.argTypes = {
+	color: {
+		control: { type: 'select', options: Object.values(BADGE_COLORS) },
+		defaultValue: BADGE_COLORS.NEUTRAL,
+	},
+};
+
+Primary.parameters = {
+	design: {
```
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:35]: IMO, there should be only one Design link per comonent, so it should be moved to `Story` definition: `const Template: Story<{ label: string; iconUrl: string; color: Object }> = (argTypes) => ({`

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 2 above)
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:13]: why `color` is `Object` type?

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 1 above)
Comment (damianmef) [reply lib/js/components/Badges/Badge/Badge.stories.ts:20]: It should be `Interactive`.

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 2 above)
Comment (damianmef) [reply lib/js/components/Badges/Badge/Badge.stories.ts:13]: Hmm so how we should name it? Like this?
```
const InteractiveStory: Story<....
...
export const Interactive = InteractiveStory.bind({});

```

Comment (kvas-damian) [reply lib/js/components/Badges/Badge/Badge.stories.ts:13]: maybe `StoryTemplate` or `AbstractStory` from OOP?

File: lib/js/components/Icon/Icon.consts.ts (hunk 3 above)
Comment (damianmef) [reply lib/js/components/Icon/Icon.consts.ts:21]: I was sure it was suggestion by IDE, but now I haven't it. Will revert it.

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 4 above)
Comment (damianmef) [reply lib/js/components/Badges/Badge/Badge.stories.ts:2]: It was suggestion in SB tutorials to add it, but it do nothing. So we can remove it.

File: lib/js/components/Buttons/Button/Button.stories.ts (hunk 8 above)
Comment (damianmef) [reply lib/js/components/Buttons/Button/Button.stories.ts:41]: I think this is unacceptable. I will try find out other way.

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 2 above)
Comment (damianmef) [reply lib/js/components/Badges/Badge/Badge.stories.ts:13]: `StoryTemplate` sounds well.

File: lib/js/components/Buttons/Button/Button.stories.ts (hunk 8 above)
Comment (damianmef) [reply lib/js/components/Buttons/Button/Button.stories.ts]: Btw. This icon in `Badge` I think we can put for now like this:
```
src="https://lek.wiecejnizlek.pl/images/lek/logo-badge.svg" 
```
this will be replaced anyway in Badge refactoring tomorrow.

Comment (damianmef) [reply lib/js/components/Buttons/Button/Button.stories.ts]: @kvas-damian I found a way, but we need discuss is it profitable to keep such a code:
```
template: `<div :class="{ contrastBackground: isInverted }">
                <Button
			v-bind="propsWithoutSlot"
                  	:icon-left="ICONS[iconLeft]"
                  	:icon-right="ICONS[iconRight]"
                >
                    {{slotText}}
                </Button>
            </div>`,
	computed: {
		propsWithoutSlot() {
			const all = Object.assign({}, this.$props);
			delete all.slotText
			return all;
		}
	},
```

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 4 above)
Comment (kvas-damian) [reply lib/js/components/Badges/Badge/Badge.stories.ts]: it's there only to tell you in which file you're, because in web code snippets you usually don't see filename

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+} as Meta<typeof Button>;
+
+const StoryTemplate: StoryFn<{
+	slotText: Object;
+	size: String;
+	type: String;
+	color: String;
+	radius: String;
+	state: String;
+	elevation: String;
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:26]: https://stackoverflow.com/a/14727461/17570299

File: lib/js/components/Buttons/Button/Button.stories.ts (hunk 8 above)
Comment (kvas-damian) [reply lib/js/components/Buttons/Button/Button.stories.ts]: it's not worth it. We can stay with the old version of passing props

File: lib/js/components/Badges/Badge/Badge.stories.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+const StoryTemplate: StoryFn<{ label: string; iconUrl: string; color: string }> = (argTypes) => ({
+	components: { Badge },
+	props: Object.keys(argTypes),
+	template:
+		'<div style="display: flex"><Badge v-bind=$props><img alt="Badge" style="width: 100%; height: 100%" :src="iconUrl" /></Badge></div>',
+	design: {
+		type: 'figma',
+		url: 'https://www.figma.com/file/izQdYyiBR1GQgFkaOIfIJI/LMS---DS---Components?node-id=1552%3A34953',
+	},
```
Comment (kvas-damian) [inline lib/js/components/Badges/Badge/Badge.stories.ts:19]:
```suggestion
```

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+import { Meta, StoryFn } from '@storybook/vue';
+
+export default {
+	title: 'Components/Buttons/Button',
+	component: Button,
+} as Meta<typeof Button>;
+
+const StoryTemplate: StoryFn<{
+	slotText: Object;
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:20]: I'm wondering if it's ok that we pass obejct with props as a type. In docs I see different approach: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args

Comment (damianmef) [reply lib/js/components/Buttons/Button/Button.stories.ts]: Hmm it wasn't simple, I need to define `Args` and `ArgTypes` types of variables, @kvas-damian check code. But it works fine.

File: lib/js/components/Badges/Badge/Badge.stories.ts (hunk 12 above)
Comment (damianmef) [reply lib/js/components/Badges/Badge/Badge.stories.ts]: Removed.

File: lib/js/components/Buttons/Button/Button.stories.ts (hunk 13 above)
Comment (damianmef) [reply lib/js/components/Buttons/Button/Button.stories.ts]: @kvas-damian my God, it looks nice but throwed me back to the problem:
Screenshot 2022-09-20 at 15 33 28

Comment (kvas-damian) [reply lib/js/components/Buttons/Button/Button.stories.ts]: is it related to the change of return type in `StoryFn`?

**Comment by damianmef** [comment]
@kvas-damian I checked a few popular Storybooks and my thoughts are:
- they are not using slots, i have never seen `slot`
- half of them was not interactive
- buttons were always filled by `label` - input text

I checked platform usages of `<wnl-button>` and for first glance = we are using slot only for text. Maybe it's a good move to change it to props? I know it's time-consuming and we don't need to do that now, but I think it's a right way. It will also help to not prescribe props now and use it as now `v-bind=$props`.
Maybe we should turning a blind eye to these ugly properties in DS?

**Comment by damianmef** [comment]
@kvas-damian I think I found a solution, in `Button.stories.ts`:
Slot as:
```
<template v-if="${'default' in args}" v-slot>${args.default}</template>
```
It has advantage - it appears as slot (screen). Disadvantage is that in html of component there is property `default="default"` but it doesn't contain slot content.
Screenshot 2022-09-27 at 18 24 26

**Comment by kvas-damian** [comment]
> Maybe it's a good move to change it to props
UX already initiated trend to avoid slots in components, bu it isn't always doable. We need them on container components, like Modal, PopOver, etc, so we need a way to tackle them

> I think I found a solution, in Button.stories.ts
I'm not sure how it solves the problem, you still get slot arg defined on html element where you use `v-bind=$props`

**Comment by kvas-damian** [comment]
I see that GitLab doesn't use `v-bind` syntax: https://gitlab.com/gitlab-org/gitlab-ui/-/blob/main/src/components/base/modal/modal.stories.js - maybe we should revert that for now?

**Comment by damianmef** [comment]
@kvas-damian sure, I revereted pros. But I think it's a good idea to stay with default slot. I think we can now close PR and think about `$props` more in the future.

File: lib/js/components/Buttons/Button/Button.stories.ts [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+	props: Object.keys(argTypes),
+	template: `<div :class="{ contrastBackground: isInverted }">
+                <Button
+					:size="size"
+					:type="type"
+					:state="state"
+					:radius="radius"
+					:color="color"
+					:elevation="elevation"
+                  	:icon-left="ICONS[$props.iconLeft]"
```
Comment (kvas-damian) [inline lib/js/components/Buttons/Button/Button.stories.ts:30]: tabs vs spaces

**Review by kvas-damian — APPROVED** [review]
